### PR TITLE
Implement logger

### PR DESCRIPTION
**Pull request by XSAM**
resolves https://github.com/open-telemetry/opentelemetry-go/issues/5061

### CODE REVIEW

File: sdk/log/record.go [hunk 1]
```diff
@@ -37,6 +37,7 @@ type Record struct {
 	// Allocation optimization: an inline array sized to hold
 	// the majority of log calls (based on examination of open-source
 	// code). It holds the start of the list of attributes.
+	// TODO: Not sure whether to use a normal array here or not, as we know the size when copy the attributes from log.Record.
```
Comment (XSAM) [inline sdk/log/record.go:40]: @open-telemetry/go-approvers, any options about this?

Should we use a normal array here? Like `attributes []log.KeyValue`

Comment (pellared) [reply sdk/log/record.go:40]: You can check the prototype for implementation.

Comment (MrAlias) [reply sdk/log/record.go:40]: > @open-telemetry/go-approvers, any options about this?
> 
> Should we use a normal array here? Like `attributes []log.KeyValue`

I think that's worth exploring, but I would do so in a follow up PR.

**Review by dashpole — APPROVED** [review]

**Review by dmathieu — APPROVED** [review]

File: sdk/log/logger.go [hunk 2]
```diff
@@ -6,28 +6,66 @@ package log // import "go.opentelemetry.io/otel/sdk/log"
...
+		if err := p.OnEmit(ctx, newRecord); err != nil {
+			otel.Handle(err)
+		}
+	}
 }
 
-func (l *logger) Enabled(ctx context.Context, r log.Record) bool {
-	// TODO (#5061): Implement.
+func (l *logger) Enabled(context.Context, log.Record) bool {
 	return true
```
Comment (MrAlias) [inline sdk/log/logger.go:43]: This needs to check all of its processors and determine if any of them will actually be enabled prior to returning true.

**Review by MrAlias — APPROVED** [review]

**Pull request merged**